stdlib/public/SDK/XCTest/XCTest.swift

```diff
@@ -1030,6 +1030,40 @@ public func XCTAssertThrowsError<T>(_ expression: @autoclosure () throws -> T, _
   }
 }
 
+public func XCTAssertNoThrow<T>(_ expression: @autoclosure () throws -> T, _ message: @autoclosure () -> String = "", file: StaticString = #file, line: UInt = #line, _ errorHandler: (_ error: Error) -> Void = { _ in }) {
+    let assertionType = _XCTAssertionType.assertion_NoThrow
+
+    // evaluate expression exactly once
+    var caughtErrorOptional: Error?
+
+    let result = _XCTRunThrowableBlock {
+        do {
+            _ = try expression()
+        } catch {
+            caughtErrorOptional = error
+        }
+    }
+
+    switch result {
+    case .success:
+        guard let caughtError = caughtErrorOptional else {
+            return
+        }
+
+        _XCTRegisterFailure(true, "XCTAssertNoThrow failed: threw error \"\(caughtError)\"", message, file, line)
```

Comment:
Shouldn't the invariant here be that if `.success` is returned from the `_XCTRunThrowableBlock` call then caughtErrorOptional should be `.none`?  It looks like `_XCTRunThrowableBlock` already handles catching Swift exceptions so you should be able to delete most of the preceding code and get down to essentially

```swift
let result = _XCTRunThrowableBlock { _ = try expression(); return }
```